apps/api-extractor/src/extractor/Extractor.ts

```diff
@@ -228,30 +243,34 @@ export class Extractor {
         if (!ApiFileGenerator.areEquivalentApiFileContents(actualApiReviewContent, expectedApiReviewContent)) {
           if (!this._localBuild) {
             // For production, issue a warning that will break the CI build.
-            this._logger.logWarning('You have changed the public API signature for this project.'
+            this._monitoredLogger.logWarning('You have changed the public API signature for this project.'
               // @microsoft/gulp-core-build seems to run JSON.stringify() on the error messages for some reason,
               // so try to avoid escaped characters:
               + ` Please overwrite ${expectedApiReviewShortPath} with a`
               + ` copy of ${actualApiReviewShortPath}`
               + ' and then request an API review. See the Git repository README.md for more info.');
           } else {
             // For a local build, just copy the file automatically.
-            this._logger.logWarning('You have changed the public API signature for this project.'
+            this._monitoredLogger.logWarning('You have changed the public API signature for this project.'
               + ` Updating ${expectedApiReviewShortPath}`);
 
             fsx.writeFileSync(expectedApiReviewPath, actualApiReviewContent);
           }
         } else {
-          this._logger.logVerbose(`The API signature is up to date: ${actualApiReviewShortPath}`);
+          this._monitoredLogger.logVerbose(`The API signature is up to date: ${actualApiReviewShortPath}`);
         }
       } else {
         // NOTE: This warning seems like a nuisance, but it has caught genuine mistakes.
         // For example, when projects were moved into category folders, the relative path for
         // the API review files ended up in the wrong place.
-        this._logger.logError(`The API review file has not been set up. Do this by copying ${actualApiReviewShortPath}`
+        this._monitoredLogger.logError(`The API review file has not been set up.`
+          + ` Do this by copying ${actualApiReviewShortPath}`
           + ` to ${expectedApiReviewShortPath} and committing it.`);
       }
     }
+
+    // If there were any errors or warnings, then fail the build
+    return (this._monitoredLogger.errorCount + this._monitoredLogger.warningCount) === 0;
   }
```

Comment:
How about return the sum of error and warning? so that we know how many errors there are.

Reply:
If the caller needs to report statistics, then I would prefer to return an actual statistics object with counts based on whatever reporting scenario there is.

However, currently I don't believe anyone has that requirement.  Either they provide a custom logger (which gives them detailed access to everything) or else they just want to know whether the build should pass or fail.

---
In reply to: [152714377](https://github.com/Microsoft/web-build-tools/pull/451#discussion_r152714377) [](ancestors = 152714377)